src/WorldStorage/WSSAnvil.cpp

```diff
@@ -2344,8 +2344,8 @@ void cWSSAnvil::LoadHorseFromNBT(cEntityList & a_Entities, const cParsedNBT & a_
 	int AgeableIdx  = a_NBT.FindChildByName(a_TagIdx, "Age");
 	if (AgeableIdx > 0)
 	{
-		Byte Age = a_NBT.GetByte(AgeableIdx);
-		Monster->SetAge(static_cast<char>(Age));
+		int Age = a_NBT.GetInt(AgeableIdx);
```

Comment:
This will now assert on maps created by the previous code, because it used to store the age as a char, but now it's reading an int. You need to detect the item's datatype and load accordingly.